sklearn/metrics/tests/test_pairwise.py

```diff
@@ -1549,3 +1549,46 @@ def test_numeric_pairwise_distances_datatypes(metric, global_dtype, y_is_x):
     dist = pairwise_distances(X, Y, metric=metric, **params)
 
     assert_allclose(dist, expected_dist)
+
+
+@pytest.mark.parametrize(
+    "X,Y,expected_distance",
+    [
+        (["a", "ab", "abc"], None, [[0.0, 1.0, 2.0], [1.0, 0.0, 1.0], [2.0, 1.0, 0.0]]),
+        (
+            ["a", "ab", "abc"],
+            [
+                "a",
+                "a",
+                "a",
+            ],
+            [[0.0, 0.0, 0.0], [1.0, 1.0, 1.0], [2.0, 2.0, 2.0]],
+        ),
+    ],
+)
+def test_pairwise_dist_custom_scoring_for_string(X, Y, expected_distance):
+    """
+    Checks the pairwise distance between any two matrices containing strings.
+    This is done by passing a metric to calculate the similarity and
+    also setting `check_length_only` to True to avoid
+    checks expecting float/int objects in input array
```

Comment:
```suggestion
    """
    Checks the pairwise distance between any two matrices containing strings.
    
    This is done by passing a metric to calculate the similarity and also setting
    `check_length_only` to True to avoid checks expecting float/int objects in input
    array.
```